### ISSUE TITLE
Problems with Travis CI/Kitchen tests

### ISSUE THREAD

**Issue by noelmcloughlin** [saltstack-formulas/mysql-formula#202]
Seems to be a problem with Travis/Kitchen setup
https://travis-ci.com/saltstack-formulas/mysql-formula/jobs/155845603
```
       ----------
                 ID: mysqld
           Function: service.running
               Name: mariadb
             Result: False
            Comment: State 'service.running' was not found in SLS 'mysql.server'
              Reason: 'service' __virtual__ returned False: No service execution module loaded: check support for service management on CentOS Linux-7
            Changes:   
       ----------
                 ID: mysql_root_password
           Function: cmd.run
               Name: mysqladmin --host "localhost" --user root password '1029694271'
             Result: False
            Comment: One or more requisite failed: mysql.server.mysqld
            Started: 22:30:33.127274
           Duration: 0.009 ms
            Changes:   
       ----------
                 ID: mysql_delete_anonymous_user_localhost.localdomain
           Function: mysql_user.absent
               Name: 
             Result: False
            Comment: One or more requisite failed: mysql.server.mysqld, mysql.server.mysql_root_password
            Started: 22:30:33.128039
           Duration: 0.01 ms
            Changes:   
       ----------
                 ID: mysql_delete_anonymous_user_localhost
           Function: mysql_user.absent
               Name: 
             Result: False
            Comment: One or more requisite failed: mysql.server.mysqld, mysql.server.mysql_root_password
            Started: 22:30:33.128377
           Duration: 0.007 ms
            Changes:   
       ----------
                 ID: mysql_delete_anonymous_user_salt-formula.ci.local
           Function: mysql_user.absent
               Name: 
             Result: False
            Comment: One or more requisite failed: mysql.server.mysqld, mysql.server.mysql_root_password
            Started: 22:30:33.128702
           Duration: 0.005 ms
            Changes:   
       ----------
                 ID: mysql_what_is_status_of_mariadb
           Function: cmd.run
               Name: service mariadb status
             Result: False
            Comment: Command "service mariadb status" run
            Started: 22:30:33.128870
           Duration: 45.52 ms
            Changes:   
              ----------
              pid:
                  685
              retcode:
                  127
              stderr:
                  /bin/bash: service: command not found
```

**Comment by noelmcloughlin** [comment]
On Centos anway.

**Comment by javierbertoli** [comment]
The issue seems to be [here](https://github.com/saltstack-formulas/mysql-formula/blob/master/mysql/server.sls#L162-L167), which seems to be a check to see if the previous [resource succeeded/failed](https://github.com/saltstack-formulas/mysql-formula/blob/master/mysql/server.sls#L144-L160) and retry it. 

It's seems the formula is using a bunch of `cmd.run` which should probably can be refactored nowadays...

**Comment by noelmcloughlin** [comment]
Thanks @javierbertoli for checking.

Yeah still happening on CentOS7 job only.
```
----------
                 ID: mysqld-service-running
           Function: service.running
               Name: mariadb
             Result: False
            Comment: State 'service.running' was not found in SLS 'mysql.server'
              Reason: 'service' __virtual__ returned False: No service execution module loaded: check support for service management on CentOS Linux-7
            Changes:   
```

**Comment by noelmcloughlin** [comment]
Fixed in #262 (cent7) and #248 (cent8)
